src/main/scala/com/microsoft/hyperspace/index/sources/FileBasedSourceProviderManager.scala

```diff
@@ -109,6 +110,45 @@ class FileBasedSourceProviderManager(spark: SparkSession) {
     }
   }
 
+  /**
+   * Returns true if the given project is a supported project. If all of the registered
+   * providers return None, this returns false.
+   *
+   * @param project Project to check if it's supported.
+   * @return True if the given project is a supported relation.
+   */
+  def isSupportedProject(project: Project, index: IndexLogEntry): Boolean = {
```

Comment:
it's more like "hasNestedColumnWithProject" ?

Reply:
I will find a better name.